app/components/collection-search.cjsx

```diff
@@ -8,40 +8,50 @@ module.exports = React.createClass
 
   getDefaultProps: ->
     multi: false
-    favorite: false
-    user: null
     project: null
 
+  getInitialState: ->
+    collections: []
+
   searchCollections: (value, callback) ->
     query =
       page_size: 10
       current_user_roles: 'owner,collaborator'
     query.search = "#{value}" unless value is ''
-    query.project_id = @props.project.id if @props.project?
 
     apiClient.type('collections').get query
       .then (collections) ->
-        opts = collections.map (collection) -> {
-          value: collection.id,
-          label: collection.display_name
-          collection: collection
-        }
+
+        opts = collections.map (collection) ->
+          {
+            value: collection.id,
+            label: collection.display_name,
+            collection: collection
+          }
 
         callback null, {
           options: opts
         }
 
-  options: ->
-    return @refs.collectionSelect.state.options
+  addSelected:(collIds) ->
```

Comment:
`react-select` gives the currently selected options as the second argument to this callback. Maybe use that instead of trying to back into it yourself?